### PR TITLE
BUG: constructors

### LINKED ISSUE #41805: BUG: Construction of Series with NAs and interval-of-datetime dtype raises error
- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

#### Code Sample, a copy-pastable example

```python
import pandas as pd

print(pd.Series(data=[None], dtype='interval[datetime64[ns]]'))

```

#### Problem description

Instantiating a `Series` with NA values in `data` fails if `dtype` is specified as `interval[datetime64[ns]]`, raising an error:

<details>

> Traceback (most recent call last):
>   File "main.py", line 3, in <module>
>     print(pd.Series(
>   File "/opt/virtualenvs/python3/lib/python3.8/site-packages/pandas/core/series.py", line 364, in __init__
>     data = sanitize_array(data, index, dtype, copy, raise_cast_failure=True)
>   File "/opt/virtualenvs/python3/lib/python3.8/site-packages/pandas/core/construction.py", line 480, in sanitize_array
>     subarr = _try_cast(data, dtype, copy, raise_cast_failure)
>   File "/opt/virtualenvs/python3/lib/python3.8/site-packages/pandas/core/construction.py", line 579, in _try_cast
>     subarr = array_type(arr, dtype=dtype, copy=copy)
>   File "/opt/virtualenvs/python3/lib/python3.8/site-packages/pandas/core/arrays/interval.py", line 280, in _from_sequence
>     return cls(scalars, dtype=dtype, copy=copy)
>   File "/opt/virtualenvs/python3/lib/python3.8/site-packages/pandas/core/arrays/interval.py", line 201, in __new__
>     return cls._simple_new(
>   File "/opt/virtualenvs/python3/lib/python3.8/site-packages/pandas/core/arrays/interval.py", line 227, in _simple_new
>     left = left.astype(dtype.subtype)
>   File "/opt/virtualenvs/python3/lib/python3.8/site-packages/pandas/core/indexes/numeric.py", line 355, in astype
>     raise TypeError(
> TypeError: Cannot convert Float64Index to dtype datetime64[ns]; integer values are required for conversion

</details>

#### Expected Output

```
0    NaN
dtype: interval
```

#### Output of ``pd.show_versions()``

<details>

> INSTALLED VERSIONS
> ------------------
> commit           : 2cb96529396d93b46abab7bbc73a208e708c642e
> python           : 3.8.10.final.0
> python-bits      : 64
> OS               : Linux
> OS-release       : 5.4.0-1043-gcp
> Version          : #46-Ubuntu SMP Mon Apr 19 19:17:04 UTC 2021
> machine          : x86_64
> processor        : x86_64
> byteorder        : little
> LC_ALL           : en_US.UTF-8
> LANG             : en_US.UTF-8
> LOCALE           : en_US.UTF-8
> 
> pandas           : 1.2.4
> numpy            : 1.20.3
> pytz             : 2021.1
> dateutil         : 2.8.1
> pip              : 19.3.1
> setuptools       : 56.0.0
> Cython           : None
> pytest           : None
> hypothesis       : None
> sphinx           : None
> blosc            : None
> feather          : None
> xlsxwriter       : None
> lxml.etree       : None
> html5lib         : 1.1
> pymysql          : None
> psycopg2         : None
> jinja2           : 3.0.1
> IPython          : None
> pandas_datareader: None
> bs4              : None
> bottleneck       : None
> fsspec           : None
> fastparquet      : None
> gcsfs            : None
> matplotlib       : 3.4.2
> numexpr          : None
> odfpy            : None
> openpyxl         : None
> pandas_gbq       : None
> pyarrow          : None
> pyxlsb           : None
> s3fs             : None
> scipy            : 1.4.1
> sqlalchemy       : 1.4.15
> tables           : None
> tabulate         : None
> xarray           : None
> xlrd             : None
> xlwt             : None
> numba            : None

</details>

### LINKED ISSUE #44917: BUG: DataFrame[interval[td64]] transpose fails
```
tdi = pd.timedelta_range("0 Days", "3 Days")
ii = pd.IntervalIndex.from_breaks(tdi)
ii = ii.insert(-1, np.nan)
df = pd.DataFrame(ii)

>>> df.T
[...]
TypeError: Cannot convert Float64Index to dtype timedelta64[ns]; integer values are required for conversion
```

This would be solved by having 2D EAs.

### LINKED ISSUE #49309: BUG: pd.Categorical(categorical, dtype="category") loses orderedness
```python
cat = pd.Categorical([3, 1], categories=[3, 2, 1], ordered=True)

res = pd.Categorical(cat, dtype="category")

>>> res.dtype
CategoricalDtype(categories=[3, 2, 1], ordered=False)
>>> cat.dtype
CategoricalDtype(categories=[3, 2, 1], ordered=True)
```

By contrast, if we do `cat.astype("category")` we _do_ retain orderedness.

Found when refactoring `sanitize_array`.  We have two code paths effectively:

```
if isinstance(data, ExtensionArray):
    if dtype is not None:
        return data.astype(dtype, copy=copy)
    elif copy:
        return data.copy()
    return data

if isinstance(dtype, ExtensionDtype):
    cls = dtype.construct_array_type()
    return cls._from_sequence(data, dtype=dtype, copy=copy)

```

Switching the order of these checks causes a few Categorical tests to break bc of the Categorical constructor behavior.

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [x] closes #49292 (Replace xxxx with the GitHub issue number)
- [x] closes #44917 (Replace xxxx with the GitHub issue number)
- [x] closes #49309 (Replace xxxx with the GitHub issue number)
- [x] closes #41805 (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/tests/indexes/datetimes/test_constructors.py [hunk 1]
```diff
@@ -37,6 +37,14 @@
 
 
 class TestDatetimeIndex:
+    def test_from_dt64_unsupported_unit(self):
+        # GH#49292
+        val = np.datetime64(1, "D")
+        result = DatetimeIndex([val], tz="US/Pacific")
+
+        expected = DatetimeIndex([val.astype("M8[s]")], tz="US/Pacific")
```
Comment (WillAyd) [inline pandas/tests/indexes/datetimes/test_constructors.py:45]: Is this actually what we expect? Any consideration to raising?

File: pandas/tests/series/test_constructors.py [hunk 2]
```diff
@@ -53,6 +53,12 @@
 
 
 class TestSeriesConstructors:
+    def test_from_na_value_and_interval_of_datetime_dtype(self):
```
Comment (WillAyd) [inline pandas/tests/series/test_constructors.py:56]: Need a whatsnew?

**Review by WillAyd — CHANGES_REQUESTED** [review]

File: doc/source/whatsnew/v2.0.0.rst [hunk 3]
```diff
@@ -747,7 +749,8 @@ Reshaping
 - Bug in :func:`join` when ``left_on`` or ``right_on`` is or includes a :class:`CategoricalIndex` incorrectly raising ``AttributeError`` (:issue:`48464`)
 - Bug in :meth:`DataFrame.pivot_table` raising ``ValueError`` with parameter ``margins=True`` when result is an empty :class:`DataFrame` (:issue:`49240`)
 - Clarified error message in :func:`merge` when passing invalid ``validate`` option (:issue:`49417`)
-- Bug in :meth:`DataFrame.explode` raising ``ValueError`` on multiple columns with ``NaN`` values or empty lists (:issue:`46084`)
+- Bug in :meth:`DataFrame.explode` raising ``ValueError`` on multiple columns with ``NaN`` values or empty lists (:issue:`46084`) Bug in :meth:`DataFrame.transpose` with ``IntervalDtype`` column with ``timedelta64[ns]`` endpoints (:issue:`44917`)
```
[the remark this answers is not in the document]
Comment (jbrockmendel) [reply doc/source/whatsnew/v2.0.0.rst:752]: copy/paste screwup, will update.

File: pandas/tests/indexes/datetimes/test_constructors.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/indexes/datetimes/test_constructors.py:45]: This is correct.  In 2.0 we support resolutions of "s", "ms", "us", and "ns", and when we get something that isn't one of those we cast to the closest supported reso

File: pandas/tests/series/test_constructors.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/tests/series/test_constructors.py:56]: i dont think so. this just adds the test, the bug got fixed sometime previously

**Review by mroeschke — APPROVED** [review]
LGTM. Merge when ready @WillAyd

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
Thanks @jbrockmendel

**Pull request merged**